activesupport/test/core_ext/time_ext_test.rb

```diff
@@ -887,13 +887,16 @@ def test_at_with_datetime_returns_local_time
 
   def test_at_with_time_with_zone
     assert_equal Time.utc(2000, 1, 1, 0, 0, 0), Time.at(ActiveSupport::TimeWithZone.new(Time.utc(2000, 1, 1, 0, 0, 0), ActiveSupport::TimeZone["UTC"]))
+    assert_equal Time.utc(2000, 1, 1, 0, 0, 0), Time.at(ActiveSupport::TimeWithZone.new(Time.utc(2000, 1, 1, 0, 0, 0), ActiveSupport::TimeZone["UTC"]), 0)
+  end
 
-    # Only test this if the underlying Time.at raises a TypeError
-    begin
-      Time.at_without_coercion(Time.now, 0)
-    rescue TypeError
```

Comment:
With `to_int`, `ActiveSupport::TimeWithZone` complies with the protocol that `Time.at` requires, and `TypeError` is no more.